File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp={}
        for i,n in enumerate(nums):
            t=target-n
            if t in mp:
                return [mp[t],i]
            mp[n]=i
        return []
        

File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    List<List<Integer>> res;
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        res=new ArrayList<>();
        solve(nums,target,0,new ArrayList<>());
        return res;
    }
    void solve(int[] nums,int t,int i,List<Integer> list){
       
            if(t==0){
                res.add(new ArrayList<>(list));
            return;
            }
            if(t<0 || i>=nums.length)return ;
        
        list.add(nums[i]);
        solve(nums,t-nums[i],i,list);
      list.remove(list.size()-1);
        solve(nums,t,i+1,list);
    }
}
